Makefile

```diff
@@ -0,0 +1,542 @@
+	sed -i '/\t_ "k8s.io\/kubernetes\/pkg\/cloudprovider\/providers\/openstack"/d' _output/ccm/pkg/cloudprovider/providers/providers.go; \
+	sed -i '/\t_ "k8s.io\/kubernetes\/pkg\/cloudprovider\/providers\/ovirt"/d' _output/ccm/pkg/cloudprovider/providers/providers.go; \
+	sed -i '/\t_ "k8s.io\/kubernetes\/pkg\/cloudprovider\/providers\/photon"/d' _output/ccm/pkg/cloudprovider/providers/providers.go; \
+	sed -i '/\t_ "k8s.io\/kubernetes\/pkg\/cloudprovider\/providers\/vsphere"/d' _output/ccm/pkg/cloudprovider/providers/providers.go
+	cd _output/ccm; \
+	tar cvfz ../ccm-vendor.tar.gz vendor pkg
+
+controller-manager: build/src artifacts/ccm-build.dockerfile _output/ccm-vendor.tar.gz
+	@echo Building the Cloud Controller Manager
+	docker build -t wrf:test -f artifacts/ccm-build.dockerfile .
+	docker run -ti --name ccm-build wrf:test
+	docker cp ccm-build:/go/src/k8s.io/kubernetes/controller-manager .
+
+_output/kubernetes-server-linux-amd64.tar.gz: _output build/kubernetes-server-linux-amd64.tar.gz controller-manager
+	tar xvfz build/kubernetes-server-linux-amd64.tar.gz -C build/server
+	cp build/server/kubernetes/LICENSES _output/server/kubernetes
+	cp -rf build/server/kubernetes/server _output/server/kubernetes
+	cp -rf build/server/kubernetes/addons _output/server/kubernetes
+	# TODO: WRF Determine if we should be propogating src tgz
+	# Add GCP specific binaries (CCM) to server tgz
+	mv controller-manager _output/server/kubernetes
+	# TODO: WRF Pull is CSI and package that with both server and kubelet.
+	cd _output/server; \
+	tar cvfz ../kubernetes-server-linux-amd64.tar.gz kubernetes
+
+_output/kubernetes-manifests.tar.gz: build _output build/kubernetes-manifests.tar.gz cluster/gce/manifests/cloud-controller-manager.manifest
+	tar xvfz build/kubernetes-manifests.tar.gz -C build/manifests
+	mkdir build/manifests/kubernetes/gci-trusty/cloud-controller-manager
+	cp cluster/gce/manifests/cloud-controller-manager.manifest build/manifests/kubernetes/gci-trusty/cloud-controller-manager
+	cp cluster/addons/cloud-controller-manager/*.yaml build/manifests/kubernetes/gci-trusty/cloud-controller-manager
+	cp cluster/gce/gci/configure-helper.sh build/manifests/kubernetes/gci-trusty/gci-configure-helper.sh
+	cd build/manifests; \
+	tar cvfz ../../_output/kubernetes-manifests.tar.gz kubernetes
+
+_output/%.tar.gz.sha512: _output/%.tar.gz
+	# Do I need to remove the binary name from the sha file?
+	sha512sum $< > $@
+
+build-artifacts: _output/kubernetes-server-linux-amd64.tar.gz _output/kubernetes-manifests.tar.gz build/kubernetes-node-linux-amd64.tar.gz build/kubernetes.tar.gz
+
+deploy-artifacts: _output/kubernetes-server-linux-amd64.tar.gz _output/kubernetes-server-linux-amd64.tar.gz.sha512 _output/kubernetes-manifests.tar.gz _output/kubernetes-manifests.tar.gz.sha512
+
+clean:
+	rm -rf build _output vendor/github.com/docker/docker controller-manager
```

Comment:
Why remove docker from vendor?

Reply:
Sorry that's a bug I had been expanding "my" vendor over the existing one and wanted to clean it up when done. No longer doing that so the clean should not have it.